common/vmexec.nim

```diff
@@ -83,17 +94,47 @@ proc valueToString(env: var VmEnv, a: VirtualAddr, typ: SemType): string =
   of tkVoid, tkError:
     unreachable()
 
-proc run*(env: var VmEnv, prc: ProcIndex, typ: SemType): string =
+proc readMemConfig*(m: VmModule): Option[MemoryConfig] =
+  ## Reads the guest memory configuration from `m` by looking for the
+  ## `stack_start`, `stack_size`, and `total_memory` global variables.
```

Comment:
I don't know how long this is going to be the case, I suspect the interface will change sooner than later, otherwise I'd suggest putting these behind an aggregate if possible to reduce name collisions

Reply:
There cannot be any name collisions, as there is no way for the source language to create exported global variables with a user-defined name. Putting them in an aggregate is also not possible, because the `L30` language only supports global variables of numeric types.

I could use a prefix for the names (e.g., `internal.stack_start`), but I think it's more likely that the approach of providing the guest memory configuration via exported globals is gone by the time user-defined globals are a thing.

Reply:
I forgot that `L30` doesn't support aggregates. I don't think it's worth making any further changes, thanks for providing more context.